tests/integration/backends/test_tpp.py

```diff
@@ -278,10 +281,32 @@ def test_clinical_events(select_all):
 
 
 @register_test_for(tpp.medications)
-def test_medications(select_all):
+@pytest.mark.parametrize(
+    "dictionary_class", [MedicationDictionary, CustomMedicationDictionary]
+)
```

Comment:
This is purely a matter of stylistic preference but, to my eyes, using parameterisation here adds an unnecessary layer of abstraction. I think it would be clearer to just add two cases (one with a MedicationDictionary entry and one with a CustomMedicationDictionary entry) and confirm they're both as expected.

In fact, you could then collapse the other separate test into this as well as a third case.

Reply:
No, the opposite! Three variations but as examples in a single test.

Reply:
Might be easier to show rather than tell ...
```py
@register_test_for(tpp.medications)
def test_medications(select_all):
    results = select_all(
        Patient(Patient_ID=1),
        # Just MedicationDictionary entry
        MedicationDictionary(MultilexDrug_ID="abc", DMD_ID="xyz"),
        MedicationIssue(
            Patient_ID=1, ConsultationDate="2020-05-15T10:10:10", MultilexDrug_ID="abc"
        ),
        # Just CustomMedicationDictionary entry
        CustomMedicationDictionary(MultilexDrug_ID="def", DMD_ID="uvw"),
        MedicationIssue(
            Patient_ID=1, ConsultationDate="2020-05-16T10:10:10", MultilexDrug_ID="def"
        ),
        # Both entries present: MedicationDictionary should be preferred
        MedicationDictionary(MultilexDrug_ID="hij", DMD_ID="rst"),
        CustomMedicationDictionary(MultilexDrug_ID="hij", DMD_ID="opq"),
        MedicationIssue(
            Patient_ID=1, ConsultationDate="2020-05-17T10:10:10", MultilexDrug_ID="hij"
        ),
    )
    assert results == [
        {
            "patient_id": 1,
            "date": date(2020, 5, 15),
            "dmd_code": "xyz",
        },
        {
            "patient_id": 1,
            "date": date(2020, 5, 16),
            "dmd_code": "uvw",
        },
        {
            "patient_id": 1,
            "date": date(2020, 5, 17),
            "dmd_code": "rst",
        },
    ]
```

Reply:
I see, thanks! I think we've traded a bit less abstraction for a bit more code; but a bit more code is okay, in tests.